### ISSUE TITLE
unexpected issues when path doesn't point to root of working tree

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#2983]
It was discovered [here](https://github.com/akuity/kargo/discussions/2967#discussioncomment-11339171) that unexpected things can happen when the `git-commit` step's `path` field points to a directory _within_ a working tree instead of to the _root_ of the working tree.

I think the manner in which `git-commit` was used here was sensible, so what I propose is:

1. When loading an existing working tree from a user-specified path, use `git rev-parse --show-toplevel` to find the root of the repo/working tree and load from there.

1. There is a rather limited number of methods defined by the `WorkTree` interface that could benefit from gaining the option to work within a path that isn't the root of the working tree. They could gain the option to specify a working directory (that is verifiably _within_ the working tree).

Between these two changes, a step like `git-commit`, when configured with a path that isn't the _root_ of the working tree, would still load the working tree properly _and_ the specified path could be used as the working directory for relevant commands.

**Comment by RohanMishra315** [comment]
Hey @krancour Can I work on this ?

**Comment by krancour** [comment]
@RohanMishra315 I assigned this to myself because I think this issue may be deceptively complex.

**Comment by RohanMishra315** [comment]
> @RohanMishra315 I assigned this to myself because I think this issue may be deceptively complex.

yeah, after carefully evaluation i thought it might require brain to do , if you have any issues I could work on than, it would be great, thanks !
